examples/tests/test_examples.py

```diff
@@ -157,37 +178,44 @@ def import_from_path(module_name, filename):
     return module
 
 
-def update_diffs(module, is_similar, img, stored_img):
+def update_diffs(module, is_similar, img, stored_img, *, atol):
     diffs_dir.mkdir(exist_ok=True)
 
-    diffs_rgba = None
-
-    def get_diffs_rgba(slicer):
-        # lazily get and cache the diff computation
-        nonlocal diffs_rgba
-        if diffs_rgba is None:
-            # cast to float32 to avoid overflow
-            # compute absolute per-pixel difference
-            diffs_rgba = np.abs(stored_img.astype("f4") - img)
-            # magnify small values, making it easier to spot small errors
-            diffs_rgba = ((diffs_rgba / 255) ** 0.25) * 255
-            # cast back to uint8
-            diffs_rgba = diffs_rgba.astype("u1")
-        return diffs_rgba[..., slicer]
+    if is_similar:
+        for path in [
+            # Keep filename in sync with the ones generated below
+            diffs_dir / f"{module}-rgb.png",
+            diffs_dir / f"{module}-alpha.png",
+            diffs_dir / f"{module}-rgb-above_atol.png",
+            diffs_dir / f"{module}-alpha-above_atol.png",
+            diffs_dir / f"{module}.png",
+        ]:
+            if path.exists():
+                path.unlink()
+        return
+
+    # cast to float32 to avoid overflow
+    # compute absolute per-pixel difference
+    diffs_rgba = np.abs(stored_img.astype("f4") - img)
+
+    diffs_rgba_above_atol = diffs_rgba.copy()
+    diffs_rgba_above_atol[diffs_rgba <= atol] = 0
+
+    # magnify small values, making it easier to spot small errors
+    diffs_rgba = ((diffs_rgba / 255) ** 0.25) * 255
+    # cast back to uint8
+    diffs_rgba = diffs_rgba.astype("u1")
+
+    diffs_rgba_above_atol = ((diffs_rgba_above_atol / 255) ** 0.25) * 255
+    diffs_rgba_above_atol = diffs_rgba_above_atol.astype("u1")
 
     # split into an rgb and an alpha diff
-    diffs = {
-        diffs_dir / f"{module}-rgb.png": slice(0, 3),
-        diffs_dir / f"{module}-alpha.png": 3,
-    }
-
-    for path, slicer in diffs.items():
-        if not is_similar:
-            diff = get_diffs_rgba(slicer)
-            iio.imwrite(path, diff)
-        elif path.exists():
-            path.unlink()
-
+    # And highlight differences that are above the atol
+    iio.imwrite(diffs_dir / f"{module}-rgb.png", diffs_rgba[..., :3])
+    iio.imwrite(diffs_dir / f"{module}-alpha.png", diffs_rgba[..., 3])
+    iio.imwrite(diffs_dir / f"{module}-rgb-above_atol.png", diffs_rgba_above_atol[..., :3])
+    iio.imwrite(diffs_dir / f"{module}-alpha-above_atol.png", diffs_rgba_above_atol[..., 3])
```

Comment:
The refactorered example screenshots help you really isolated the differences that we consider "bad". In the case  of the one point marker test that is failing, the background gradient is creating many errors that "delta = 1", but the one pixel that is locally causing a delta of 2 is located:
![image](https://github.com/user-attachments/assets/9a2c7a6b-16de-4c7d-b10a-a3d8c0850058)